src/components/Dialog/Dialog.js

```diff
@@ -0,0 +1,138 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { MdClose } from 'react-icons/md';
+import Draggable from 'react-draggable';
+import Header from '../Header';
+import Button from '../Button';
+
+const propTypes = {
+  /**
+   * Children content of the Dialog.
+   */
+  children: PropTypes.element,
+
+  /**
+   * CSS class of the dialog.
+   */
+  className: PropTypes.string,
+
+  /**
+   * CSS class of the dialog's children.
+   */
+  classNameChildren: PropTypes.string,
+
+  /**
+   * CSS class of the dialog's header.
+   */
+  classNameHeader: PropTypes.string,
+
+  /**
+   * CSS class of the dialog close button.
+   */
+  classNameCloseBt: PropTypes.string,
+
+  /**
+   * Function triggered on closing dialog.
+   */
+  onClose: PropTypes.func.isRequired,
+
+  /**
+   * If set to true, allow user to drag the dialog.
+   */
+  isDraggable: PropTypes.bool,
+
+  /**
+   * If set to true, blocks the application, and disable dragging action.
+   */
+  isModal: PropTypes.bool,
+
+  /**
+   * If set to true, dialog is displayed.
+   */
+  isOpen: PropTypes.bool,
+
+  /**
+   * Dialog title.
+   */
+  title: PropTypes.element,
+};
+
+const defaultProps = {
+  children: undefined,
+  className: undefined,
+  classNameChildren: undefined,
+  classNameHeader: 'tm-dialog-header',
+  classNameCloseBt: 'tm-button tm-dialog-close-bt',
+  isDraggable: false,
+  isModal: false,
+  isOpen: false,
+  title: undefined,
+};
+
+/**
+ * This component creates a Dialog window.
+ */
+class Dialog extends Component {
+  renderDialogTitle() {
+    const {
+      title,
+      onClose,
+      classNameHeader,
+      classNameCloseBt,
+      isDraggable,
+    } = this.props;
+    return (
+      <Header
+        className={`${classNameHeader} ${
+          isDraggable ? 'tm-dialog-draggable' : ''
+        }`}
+      >
+        {title}
+        <Button className={classNameCloseBt} onClick={() => onClose()}>
+          <MdClose focusable={false} />
+        </Button>
+      </Header>
+    );
+  }
+
+  renderDialog() {
+    const {
+      className,
+      classNameChildren,
+      children,
+      isModal,
+      isOpen,
+    } = this.props;
+    return (
+      <div
+        className={isModal ? 'tm-modal' : 'tm-dialog'}
+        style={{ display: isOpen ? 'block' : 'none' }}
+      >
+        <div
+          className={`${
+            isModal ? 'tm-modal-container' : 'tm-dialog-container'
```

Comment:
you don't need  to differentiate the container className, because the parent already define if it's a modal or not.

Reply:
But it needs a different style however (see scss file)